src/MSBuild/XMake.cs

```diff
@@ -1142,7 +1143,14 @@ bool isolateProjects
 
                             if (!restoreOnly)
                             {
-                                results = ExecuteBuild(buildManager, request);
+                                if (isolateProjects)
```

Comment:
Hmm, this is a little strange. I still think we'll eventually need 2 separate flags. One for isolated builds and one for graph builds. Isolated means without graph means 1 single project and import the build result state from its references somehow, and a graph build could either be isolated or not (basically use graph scheduling with or without the isolation enforcement). For now I think using this one flag for everything is OK until we start needing to disambiguate the scenarios (we'll start with just graph + isolated both on)